comp_arguments.go

```diff
@@ -0,0 +1,123 @@
+package mcli
+
+import (
+	"context"
+)
+
+/*
+Arg completion tag syntax (WIP):
+
+	`comp:"nofile"`
+	`comp:"enums=a,b,c,d;"`
+	`comp:"fn=funcName"`
+	`comp:"listExt=json,yaml,toml"`
+	`comp:"listFiles"`  // and listDirs, listDirsAndFiles, etc.
+
+*/
```

Comment:
With more looking into this feature, I am thinking that adding a new tag is not the best solution, maybe simply use a parsing option is a simpler and better choice. e.g.

```
WithArgCompFuncs(funcs map[string]ArgCompletionFunc) ParseOpt

// where key of funcs is a flag name "-a", or "--b", or an argument name "arg1"
```

Just a thought, how do you think?

Reply:
Could you prepare example?

Reply:
e.g.
```
func main() {
	var args struct {
		Name string `cli:"-n, --name, Who do you want to say to" default:"tom"`

		// This argument is required.
		Text string `cli:"#R, text, The 'message' you want to send"`
	}
	mcli.Parse(&args,
		mcli.WithArgCompFuncs(map[string]mcli.ArgCompletionFunc{
			"-n": func(ctx mcli.ArgCompletionContext) []mcli.CompletionItem {
				return []mcli.CompletionItem{
					{"Tom", "Tom Smith"},
					{"John", "John White"},
				}
			},
			"text": func(ctx mcli.ArgCompletionContext) []mcli.CompletionItem {
				// biz logic
				// ...
				return []mcli.CompletionItem{
					// ...
				}
			},
		}))
	fmt.Printf("Say to %s: %s\n", args.Name, args.Text)
}
```

Reply:
ok, cleanup done
looks actually better